lib/ansiblelint/rules/PackageHasRetryRule.py

```diff
@@ -76,12 +76,37 @@ class PackageHasRetryRule(AnsibleLintRule):
         "absent",
     ]
 
+    _package_name_keys = [
+        "name",
+        "package",
+        "pkg",
+        "deb",
+    ]
+
     def matchtask(self, file, task):
         module = task["action"]["__ansible_module__"]
-        if module in self._package_modules:
-            if 'until' not in task:
-                # Exclude tasks where state is `absent`.
-                if task['action'].get('state') not in self._module_ignore_states:
-                    message = '{0} {1}'
-                    return message.format(self.shortdesc, module)
+
+        if module not in self._package_modules:
+            return False
+
+        is_task_retryable = 'until' in task
+        if is_task_retryable:
+            return False
+
+        is_state_whitelisted = task['action'].get('state') in self._module_ignore_states
+        if is_state_whitelisted:
+            return False
+
+        # attempt to check if this is a local package file
+        for key in self._package_name_keys:
+            found_package_name = task['action'].get(key)
+            if found_package_name:
+                break
+        if found_package_name:
```

Comment:
Flip this to
```suggestion
        if not found_package_name:
            return True
```
to maintain low nesting.